### DISCUSSION TITLE
Separate "type" from "content"

### DISCUSSION THREAD

**Discussion by Jay-o-Way** [U-C-S/Hurl#166] [Ideas]
Idea:  do not build the _type_ into the _rule_ string, but let this be a separate field in the JSON. This can make handling of the code easier, and removes the prefix, so users don't have to deal with this (read: can't make mistakes). I think in the UI a ComboBox would be great, because it makes it very easy to set (bind with **RuleMode**).

Example:

```json
"Rule" {
    "Type": "domain",
    "Content": "github.com"
}
```

> [!NOTE]
> This means omitting the prefix will no longer be possible and therefore no longer allowed. (looking at you, _schema_)

**Comment by U-C-S** [comment]
I prefer the way it is currently since i believe it's easier to write a rule simply by opening the file in a text editor without the `,` comma errors. 

While I am open to change it, I am in ambiguity

We are transforming it anyway here: https://github.com/U-C-S/Hurl/blob/main/Source/Hurl.Library/Models/Ruleset.cs

**Reply by Jay-o-Way** [reply]
Yes, more user input is welcome here. I wonder how many people use a text editor to change settings like this, versus the UI.
